novawallet/Common/Services/PushNotificationsSettingsService/PushSettings.swift

```diff
@@ -0,0 +1,82 @@
+import FirebaseCore
+import FirebaseFirestore
+import SubstrateSdk
+
+typealias ChainSelection = PushSettings.Selection<[String]>
+
+struct PushSettings: Codable, Equatable {
+    struct Wallet: Codable, Equatable {
+        let baseSubstrate: AccountAddress
+        let baseEthereum: AccountAddress?
+        let chainSpecific: [String: String]
+    }
+
+    struct Notifications: Codable, Equatable {
+        let announcements: Bool
```

Comment:
I don't think we need to store announcements flag on the Firestore. It is stored only locally. Probably better to have a separate remote model.